src/debug.rs

```diff
@@ -23,6 +26,30 @@ pub trait DebugQueryTable {
     fn keys<C>(&self) -> C
     where
         C: FromIterator<Self::Key>;
+
+    /// Get the (current) set of the entries in the query table.
+    fn entries<C>(&self) -> C
+    where
+        C: FromIterator<TableEntry<Self::Key, Self::Value>>;
+}
+
+/// An entry from a query table, for debugging and inspecting the table state.
+pub struct TableEntry<K, V> {
+    /// key of the query
+    pub key: K,
+    /// value of the query, if it is stored
+    pub value: Option<V>,
+    _for_future_use: (),
```

Comment:
We probably want to expose `is_const`, `verified_at`, etc here, but, as I don't have immediate need for those, I've ommited them.